test/e2e/agent/agent_updates_test.go

```diff
@@ -75,6 +75,87 @@ var _ = Describe("VM Agent behavior during updates", func() {
+				newImageReference = repo + ":v4"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Update")
+				}, "1m")
+
+			harness.WaitForDeviceContents(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Rebooting")
+				}, "2m")
+
+			harness.WaitForDeviceContents(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference &&
+						conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// related to: https://issues.redhat.com/browse/EDM-679
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			logrus.Infof("Device updated to new image %s 🎉", "flightctl-device:v4")
+			logrus.Info("Pods with sleep infinity process are present but not running, as expected")
+			stdout, err := harness.VM.RunSSH([]string{"sudo", "podman", "ps"}, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(ContainSubstring("sleep infinity"))
+
+			logrus.Info("We expect containers with sleep infinity process to be present but not running 👌")
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":base"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 3",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Update")
+				}, "1m")
+
+			harness.WaitForDeviceContents(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Rebooting")
+				}, "2m")
+
+			harness.WaitForDeviceContents(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference &&
+						conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+
+			logrus.Infof("Device updated to new image %s 🎉", "flightctl-device:base")
+			Expect(device.Spec.Applications).To(BeNil())
+			logrus.Info("Application demo_embedded_app is not present in new image 🌞")
+
+			stdout1, err1 := harness.VM.RunSSH([]string{"sudo", "podman", "ps"}, nil)
```

Comment:
optional: this is just a style nit but its ok to reuse variables given proper scoping.
```suggestion
			stdout, err = harness.VM.RunSSH([]string{"sudo", "podman", "ps"}, nil)
```

Reply:
The linter doesnt allow it

Reply:
note
`stdout, err =`  vs `stdout, err =:`